packages/astro/src/@types/astro.ts

```diff
@@ -1010,6 +1012,71 @@ export interface AstroUserConfig {
 		 * ```
 		 */
 		service: ImageServiceConfig;
+
+		/**
+		 * @docs
+		 * @name image.domains (Experimental)
+		 * @type {Array<string>}
+		 * @default `{domains: []}`
+		 * @version 2.9.2
+		 * @description
+		 * Defines a list of domains allowed to be optimized locally
+		 *
+		 * The value should be an array of domain strings of images allowed to be optimized locally.
+		 * Wildcards are not permitted (see image.remotePatterns).
+		 *
+		 * ```js
+		 * {
+		 *   image: {
+		 *     // Example: Enable the Sharp-based image service
+		 *     domains: ['astro.build'],
+		 *   },
+		 * }
+		 * ```
+		 */
+		domains?: string[];
+
+		/**
+		 * @docs
+		 * @name image.remotePatterns (Experimental)
+		 * @type {Array<RemotePattern>}
+		 * @default `{remotePatterns: []}`
+		 * @version 2.9.2
+		 * @description
+		 * Defines a list of url patterns allowed to be optimized locally
+		 *
+		 * The patterns can contain 4 properties:
+		 * 1. protocol
+		 * 2. hostname
+		 * 3. port
+		 * 4. pathname
+		 *
+		 * Each property will be matched individually to the URL object representing the remote asset to be loaded.
+		 * The rules for matching are:
+		 *
+		 * 1. protocol & port: strict equal.(`===`)
+		 * 2. hostname:
+		 *   - if the hostname starts with '**.', all subdomains will be allowed ('endsWith')
+		 *   - if the hostname starts with '*.', only one level of subdomain will be allowed
+		 *   - strict equal otherwise
+		 * 3. pathname:
+		 *   - if the pathname ends with '/**', all sub routes will be allowed ('startsWith')
+		 *   - if the pathname ends with '/*', only one level of sub route will be allowed
+		 *   - strict equal otherwise
+		 *
+		 * ```js
+		 * {
+		 *   image: {
+		 *     // Example: statically process all images from your aws s3 bucket
+		 *     remotePatterns: [{
+		 *       protocol: 'https',
+		 *       hostname: '**.amazonaws.com',
+		 *     }],
+		 *   },
+		 * }
+		 * ```
+		 */
```

Comment:
``````suggestion
		 * ```js
		 * {
		 *   image: {
		 *     // Example: statically process all images from your aws s3 bucket
		 *     remotePatterns: [{
		 *       protocol: 'https',
		 *       hostname: '**.amazonaws.com',
		 *     }],
		 *   },
		 * }
		 * ```
		 * 
		 * You can use wildcards to define the permitted `hostname` and `pathname` values as described below. Otherwise, only the exact values provided will be configured:
		 * `hostname`:
		 *   - Start with '**.' to allow all subdomains ('endsWith').
		 *   - Start with '*.' to allow only one level of subdomain.
		 * 
		 * `pathname`:
		 *   - End with '/**' to allow all sub-routes ('startsWith').
		 *   - End with '/*' to allow only one level of sub-route.
		 */
``````
I think showing the example first is more helpful, so I've swapped these. Also, I'm not sure, but I seem to see "subroute`"more often as one word or hyphenated when I look it up. Making it one word here gave me red squiggles, but a hyphen didn't. So, arbitrarily, I'm suggesting that one. But if you know for sure, then choose that one.

Also, the reader is probably less interested in reading this from a "what your feature's matching patterns are" standpoint as they are in knowing *how to configure* for what they want. So I've made a suggestion where you can see what framing it this way this looks like. Bonus, lines are shorter, more direct, and are instructions to the users.